src/devices/cpu/arcompact/arcompact_execute.cpp

```diff
@@ -1253,18 +1253,37 @@ ARCOMPACT_RETTYPE arcompact_device::arcompact_handle00_01(OPS_32)
 
 }
 
-ARCOMPACT_RETTYPE arcompact_device::arcompact_handle01_00_00dasm(OPS_32)
+ARCOMPACT_RETTYPE arcompact_device::arcompact_handle01_00_00(OPS_32)
 {
 	int size = 4;
 
 	// Branch and Link Conditionally
-	arcompact_log("unimplemented BLcc %08x", op);
+	// 0000 1sss ssss sss0 SSSS SSSS SSNQ QQQQ
+	int32_t address = (op & 0x07fe0000) >> 17;
+	address |= ((op & 0x0000ffc0) >> 6) << 10;
+	if (address & 0x80000) address = -0x80000 + (address & 0x7ffff);
+	int n = (op & 0x00000020) >> 5; op &= ~0x00000020;
```

Comment:
Please clean up the line breaks – having multiple statements on a single line isn’t making things clearer here.

Reply:
The code was already like this.

Reply:
The code already being like that is part of the reason why Haze said that it might be better to start the core fresh. If your intent is to improve the core, cleaning up questionable formatting is part of that responsibility.